queue/src/fiskaltrust.Middleware.Contracts/RequestCommands/ZeroReceiptCommand.cs

```diff
@@ -50,28 +65,39 @@ public override async Task<RequestCommandResponse> ExecuteAsync(ftQueue queue, R
             var sentReceipts = new List<string>();
             var signatures = new List<SignaturItem>();
 
+            var succeeded = true;
             if (_resendFailedReceipts)
             {
-                await ResendFailedReceiptsAsync(iQueue, queue, sentReceipts, signatures).ConfigureAwait(false);
+                succeeded = await ResendFailedReceiptsAsync(iQueue, queue, sentReceipts, signatures).ConfigureAwait(false);
+            }
+
+            var resent = $"Resent {sentReceipts.Count()} receipts that have been stored between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G}.";
+
+            if (succeeded && signingAvailable)
+            {
+                _logger.LogInformation($"Successfully closed failed-mode. {resent} ");
+                iQueue.SSCDFailCount = 0;
+                iQueue.SSCDFailMoment = null;
+                iQueue.SSCDFailQueueItemId = null;
+            }
+            else
+            {
+                receiptResponse.ftState = _countryBaseState | 2;
             }
             receiptResponse.ftStateData = JsonConvert.SerializeObject(new { SentReceipts = sentReceipts });
-            _logger.LogInformation($"Successfully closed failed-mode, resent {sentReceipts.Count()} receipts that have been stored between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G}.");
 
-            var caption = $"Restored connection to fiscalization service at {DateTime.UtcNow:G}.";
-            var data = $"{iQueue.SSCDFailCount} receipts from the timeframe between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G} have been re-processed at the fiscalization service.";
+            _logger.LogInformation($"{sentReceipts.Count()} receipts from the timeframe between {iQueue.SSCDFailMoment:G} and {DateTime.UtcNow:G} have been re-processed at the fiscalization service.");
+
+            var stateDetail = JsonConvert.SerializeObject(new StateDetail() { FailedReceiptCount = iQueue.SSCDFailCount, FailMoment = iQueue.SSCDFailMoment, SigningDeviceAvailable = signingAvailable });
 
             signatures.Add(new()
             {
                 ftSignatureType = _countryBaseState | 2,
                 ftSignatureFormat = (long) ifPOS.v0.SignaturItem.Formats.Text,
-                Caption = caption,
-                Data = data
+                Caption = stateDetail,
+                Data = JsonConvert.SerializeObject(new { SentReceipts = sentReceipts })
             });
```

Comment:
If I understood @TSchmiedlechner correctly we can actually completely remove this signatureItem as this does not need to be printed on the receipt in italy 🤔

Reply:
I think that also means that we could potentially have left out some of those fields in the IT storage 🤔 we can think about that in the next market ^^